### PR TITLE
Retry buildkite pipelines for nightly tests + on-commit pipelines except the ones in the diff plugin

### PR DESCRIPTION
**Pull request by artem-shelkovnikov**
Properly implement retries for failing buildkite pipelines: before they did not work at all.

All commits you see here were to test what works.

What works:
- Retrying all steps in nightly pipelines
- Retrying steps that are not a part of `diff` plugin

This change should reduce flakyness of our pipelines

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -522,7 +526,6 @@ steps:
             - "requirements/**"
             config:
               label: "🔨 GraphQL"
-              <<: *test-agents
```
Comment (navarone-feekery) [inline .buildkite/pipeline.yml:525]: Did you intend to have `*retries` added here too?

Comment (artem-shelkovnikov) [reply .buildkite/pipeline.yml:525]: Oh wow I also did not intend to delete the test-agents here, good catch!

**Review by navarone-feekery — APPROVED** [review]
LGTM!

File: .buildkite/nightly_steps.yml [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+    retry:
+      automatic:
+        - exit_status: -1  # Connection to the Agent was lost
+          signal_reason: none
+          limit: 2
+        - exit_status: 255  # Timeout
+          signal_reason: none
+          limit: 2
+        - exit_status: 2  # Flaky test
+          signal_reason: none
```
Comment (navarone-feekery) [inline .buildkite/nightly_steps.yml:12]: Late non-blocking question: `signal_reason: none` wasn't originally included for the flaky test retry. What changes by adding this?

Comment (artem-shelkovnikov) [reply .buildkite/nightly_steps.yml:12]: It's here for consistency.

`signal_reason` itself can be [one of these options](https://buildkite.com/docs/pipelines/command-step): 

- "*"
- none
- cancel
- agent_stop

For now it seems like retrying on "cancel" or "agent_stop" does not make sense, so retrying only when there's no reason.

**Pull request merged**